### DISCUSSION TITLE
Mutiny as reactive streams binding for subscipriton

### DISCUSSION THREAD

**Discussion by guykatz** [graphql-java/graphql-java#2662] [General]
Hi;
I am asking out of curouasity.
in the current impl graphql-java is using either rx java or project reactor as implementations for the reactive streams part of subscirptions.
recently I have bumped into smallrye Mutiny. I was wondering: would anyone consider adding Mutiny as a binding? 
thanks!

**Comment by andimarek** [comment]
Hi,

GraphQL Java doesn't use any specific reactive implementation. The only dependency is [Reactive Streams](https://www.reactive-streams.org/) which is a general reactive spec.

**Reply by bbakerman** [reply]
To expand more - graphql-java aims to have as few dependencies as possible.  So we sought to use the reference reactive streams as a way to have a generalised interface for "streams of data" that subscriptions produce.  

From the Mutiny site

> Mutiny is based on the Reactive Streams specification, and so it can be integrated with any other reactive programming library.

So you can adapt the stream of data that graphql-java subscriptions produce to Mutiny code
